src/EIP7702Proxy.sol

```diff
@@ -123,6 +123,7 @@ contract EIP7702Proxy is Proxy {
         // block guarded initializer from being called
         if (msg.sig == guardedInitializer) revert InvalidInitializer();
 
+        /// @review should we prevent all delegate calls if the implementation is not yet initialized? Feels like a good default protection
```

Comment:
https://github.com/base-org/eip-7702-proxy/pull/9/files#r1936136639  I'm not sure it's worth that check here given that I don't see the security gain and it'd require a conditional check on every delegatecall